ui/app/src/components/CompareVersionsDialog/FieldVersionToolbar.tsx

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (C) 2007-2024 Crafter Software Corporation. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as published by
+ * the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import Button from '@mui/material/Button';
+import ChevronLeftRoundedIcon from '@mui/icons-material/ChevronLeftRounded';
+import Paper from '@mui/material/Paper';
+import Box from '@mui/material/Box';
+import Typography from '@mui/material/Typography';
+import Divider from '@mui/material/Divider';
+import IconButton from '@mui/material/IconButton';
+import TextSnippetOutlinedIcon from '@mui/icons-material/TextSnippetOutlined';
+import CodeOutlinedIcon from '@mui/icons-material/CodeOutlined';
+import NavigateNextRoundedIcon from '@mui/icons-material/NavigateNextRounded';
+import React, { ReactNode } from 'react';
+import { ContentTypeField } from '../../models';
+import Tooltip from '@mui/material/Tooltip';
+import { FormattedMessage } from 'react-intl';
+import NotesRoundedIcon from '@mui/icons-material/NotesRounded';
+import { useHotkeys } from 'react-hotkeys-hook';
+
+interface FieldVersionToolbarProps {
+  field: ContentTypeField;
+  contentTypeFields: ContentTypeField[];
+  compareXml: boolean;
+  showCleanText: boolean;
+  actions?: ReactNode;
+  showFieldsNavigation?: boolean;
+  setCompareXml(value: boolean): void;
+  setShowCleanText(value: boolean): void;
+  onSelectField?(field: ContentTypeField): void;
+}
+
+export function FieldVersionToolbar(props: FieldVersionToolbarProps) {
+  const {
+    field,
+    contentTypeFields,
```

Comment:
The contentType and the fields is a good candidate for context.
Similarly the selectedField, could be something that both the side bar and the single field view could leverage in context